File: two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        idx_map = {}
        
        for i, val in enumerate(nums):
            num = target - val
            
            if num in idx_map:
                return [i, idx_map[num]]
            
            idx_map[val] = i

File: two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    const idxMap = {};
    
    for (let i = 0; i < nums.length; i++) {
        let num = nums[i]
        let val = target - num;
        
        if (val in idxMap) {
            return [i, idxMap[val]]
        }
        
        idxMap[num] = i;
    }
};
